### PR TITLE
Add persistence layers to the application #2

### PR DESCRIPTION
**Pull request by matteosz**
This PR follows #1528 to address the issue #1383. 

In details, here I implemented a persistence layer for:
- ElectionRepository
- RollcallRepository

Also, there's an improvement fix for some tests and the disposables of the asynchronous calls.

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    EventState state = stateString.isEmpty() ? null : EventState.valueOf(stateString);
+
+    // Deserialize collections
+    JsonArray electionQuestionsJsonArray = jsonObject.get("electionQuestions").getAsJsonArray();
+    List<ElectionQuestion> electionQuestions = new ArrayList<>();
+    for (JsonElement electionQuestionJsonElement : electionQuestionsJsonArray) {
+      electionQuestions.add(
+          context.deserialize(electionQuestionJsonElement, ElectionQuestion.class));
+    }
+
```
Comment (simone-kalbermatter) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java:47]: You could add a comment about what is being deserialized in the codeblocks below as you did for the previous ones.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        results);
+  }
+
+  @Override
+  public JsonElement serialize(
+      Election election, Type typeOfSrc, JsonSerializationContext context) {
+    JsonObject jsonObject = new JsonObject();
+
+    jsonObject.add("channel", context.serialize(election.getChannel(), Channel.class));
+
```
Comment (simone-kalbermatter) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java:102]: Like in `deserialize`, you could add a comment about what is being serialized in each code block, it makes the code easier to read.

**Review by simone-kalbermatter — CHANGES_REQUESTED** [review]
LGTM, just some small comments.

**Review by simone-kalbermatter — APPROVED** [review]
LGTM!

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusAcceptTest.java [hunk 3]
```diff
@@ -28,11 +26,6 @@ public class ConsensusAcceptTest {
   private static final ConsensusAccept accept =
       new ConsensusAccept(instanceId, messageId, timeInSeconds, acceptedTry, acceptedValue);
 
-  @Before
-  public void setup() {
-    JsonTestUtils.loadGSON(ApplicationProvider.getApplicationContext());
-  }
-
```
Comment (Warhedgehog) [inline fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusAcceptTest.java:35]: I'm curious, why isn't it needed anymore ?

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Great work, some minor things

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusAcceptTest.java (hunk 3 above)
Comment (matteosz) [reply fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusAcceptTest.java:35]: In the first place I put the setup method as the loadGson was using the application context, then I realised that I didn't need the context there, so it was better to have a static centralised initialisation over than multiple setups. This was also generating multiple instances of databases so not being optimal

**Review by Warhedgehog — APPROVED** [review]
LGTM

**Pull request merged**